cmd/common/snapshot/process.go

```diff
@@ -35,6 +40,90 @@ type ProcessParser struct {
 	outputConfig *commonutils.OutputConfig
 }
 
+type nodeProcess struct {
+	event    *types.Event
+	children []*nodeProcess
+}
+
+func createTree(processes []*types.Event) (*nodeProcess, error) {
+	var root *nodeProcess
+
+	nodes := make(map[int]*nodeProcess, len(processes))
+	// Create a node for each processes.
+	for _, process := range processes {
+		nodes[process.Pid] = &nodeProcess{
+			event:    process,
+			children: make([]*nodeProcess, 0),
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		ppid := node.event.ParentPid
+		if _, ok := nodes[ppid]; !ok {
```

Comment:
Is it safe to assume that there will be exactly one process that doesn't have an entry for its parent - if the iterator also works by iterating over a process tree, everything should be fine, but I don't know if that's the case.